plugins/google-sheets/src/App.tsx

```diff
@@ -47,9 +47,13 @@ export function AuthenticatedApp({ pluginContext, setContext }: AuthenticatedApp
     const [spreadsheetId, setSpreadsheetId] = useState<string | null>(
         pluginContext.type === "update" ? pluginContext.spreadsheetId : null
     )
+    const [sheetId, setSheetId] = useState<number | null>(
+        pluginContext.type === "update" ? pluginContext.sheetId : null
+    )
     const [sheetTitle, setSheetTitle] = useState<string | null>(
         pluginContext.type === "update" ? pluginContext.sheetTitle : null
     )
+
```

Comment:
Need to do here the same thing as in `syncSheet` and use `sheetTitleFromId` by default, and fall back to `sheetTitle` only if that one is `null`. This is the source of the bug where "Manage" doesn't work on an "upgraded" collection with a renamed sheet.